backend/src/auth/auth.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { PersonnelEntity } from 'src/database/entities/personnel.entity';
+import { Program, RequestWithRoles, Role, Token } from './interface';
+import { AppLogger } from '../logger/logger.service';
+import { PersonnelService } from '../personnel/personnel.service';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @Inject(PersonnelService)
+    private readonly personnelService: PersonnelService,
+    private readonly logger: AppLogger,
+  ) {
+    this.logger.setContext(AuthService.name);
+  }
+
+  /**
+   * Set the usersname and the role to the request object
+   * The Coordinator role has the highest permissions level, so if the user has the Coordinator role, the role will be set to Coordinator.
+   * The logistics role has the second highest permissions level, so if the user has the logistics role, the role will be set to logistics.
+   * If a user has both roles the role will be set to Coordinator.
+   * @param payload The token payload
+   * @param request The request object
+   */
+  async setRequestUserInfo(payload: Token, request: Request): Promise<boolean> {
+    if (payload.given_name && payload.family_name) {
+      request['username'] = `${payload.given_name} ${payload.family_name}`;
+    } else {
+      request['username'] = '';
+    }
+    if (payload.email) {
+      request['idir'] = payload.email;
+    } else {
+      request['idir'] = '';
+    }
+
+    const isMember = await this.personnelService.verifyMember(payload.email);
```

Comment:
Can shorten this to a single request, get all personnel with email or supervisor as this email. Less db requests the better IMO.

Reply:
I agree with this in general, but I think it is more readable this way?

Reply:
Otherwise we're querying where email === email or where email === supervisor email and then having to figure out whether it was the email that matched or the supervisor email that matched. I'm just trying to keep the role setting as clear as possible since it's part of the auth guard. But open to discussion!